src/components/ModalDialog/ModalDialog.vue

```diff
@@ -0,0 +1,282 @@
+        <slot></slot>
+      </section>
+      <section
+        v-if="primaryButtonText"
+        :class="[
+          'unnnic-modal-dialog__container__actions',
+          showActionsDivider &&
+            'unnnic-modal-dialog__container__actions--divider',
+        ]"
+      >
+        <UnnnicButton
+          type="tertiary"
+          :text="secondaryButtonText || i18n('cancel')"
+          @click.stop="
+            secondaryButtonText ? $emit('secondaryButtonClick') : close()
+          "
+        />
+        <UnnnicButton
+          :type="
+            primaryButtonType || primaryButtonTypeMapper[type] || 'primary'
+          "
+          :text="primaryButtonText"
+          @click.stop="$emit('primaryButtonClick')"
+        />
+      </section>
+    </section>
+  </section>
+</template>
+
+<script>
+import UnnnicIcon from '../Icon.vue';
+import UnnnicButton from '../Button/Button.vue';
+import UnnnicI18n from '../../mixins/i18n';
+
+export default {
+  name: 'UnnnicModalDialog',
+  components: {
+    UnnnicIcon,
+    UnnnicButton,
+  },
+  mixins: [UnnnicI18n],
+  props: {
+    modelValue: {
+      type: Boolean,
+      required: true,
+    },
+    persistent: {
+      type: Boolean,
+      default: false,
+    },
+    type: {
+      type: String,
+      default: '',
+      validate(type) {
+        return ['success', 'warning', 'attention'].includes(type);
+      },
+    },
+    size: {
+      type: String,
+      default: 'md',
+      validate(size) {
+        return ['sm', 'md', 'lg'].includes(size);
+      },
+    },
+    title: {
+      type: String,
+      default: 'title',
+    },
+    icon: {
+      type: String,
+      default: '',
+    },
+    iconScheme: {
+      type: String,
+      default: '',
+    },
+    showCloseIcon: {
+      type: Boolean,
+      default: false,
+    },
+    primaryButtonText: {
+      type: String,
+      default: '',
+    },
+    secondaryButtonText: {
+      type: String,
+      default: '',
+    },
+    primaryButtonType: {
+      type: String,
+      default: '',
+    },
+    showActionsDivider: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  emits: ['primaryButtonClick', 'secondaryButtonClick', 'update:modelValue'],
+
+  data() {
+    return {
+      defaultTranslations: {
+        cancel: {
+          'pt-br': 'Cancelar',
+          en: 'Cancel',
+          es: 'Cancelar',
+        },
+      },
+      iconsMapper: {
+        success: { icon: 'check_circle', scheme: 'aux-green-500' },
+        warning: { icon: 'warning', scheme: 'aux-red-500' },
+        attention: { icon: 'error', scheme: 'aux-yellow-500' },
+      },
+      primaryButtonTypeMapper: {
+        success: 'primary',
+        warning: 'warning',
+        attention: 'attention',
+      },
+    };
+  },
+  watch: {
+    modelValue(value) {
+      if (value) document.body.style.overflow = 'hidden';
+      else document.body.style.overflow = '';
+    },
+  },
```

Comment:
Firstly, I recommend removing the logic from within the modelValue watcher, replacing it with a function that explains what is being done, as in the example below. This improves the readability and maintainability of the code, avoiding that, when adding other logic to this watcher, reading the contexts within it becomes difficult.

```suggestion
  watch: {
    modelValue(value) {
      this.updateBodyOverflow(value);
    },
  },
```
Secondly, within the `updateBodyOverflow` function, I would choose to use the ternary condition, since in this case it becomes more readable than an if-else (mainly inline), as in the example below:

```
updateBodyOverflow(isHidden) {
  document.body.style.overflow = isHidden ? 'hidden' : '';
},
```